add --verify option

jayteemo opened the pull request:
Add `--verify` option which verifies that the device has credentials in each slot of the sec tag and that the SHA matches what was written.
A feature request from @junqingzou

jayteemo committed:
add --verify option

glarsennordic commented on python/modem-firmware-1.3+/device_credentials_installer.py:
```diff
@@ -699,5 +712,63 @@ def main():
     cleanup()
 
+def verify_credentials(sec_tag, ca_cert, client_cert):
+    write_line('AT%CMNG=1,{},0'.format(sec_tag))
```
Some comments annotating the intent of each AT command would probably be useful

glarsennordic commented on python/modem-firmware-1.3+/device_credentials_installer.py:
```diff
@@ -699,5 +712,63 @@ def main():
 
+def verify_credentials(sec_tag, ca_cert, client_cert):
+    write_line('AT%CMNG=1,{},0'.format(sec_tag))
+    retval, ca_res = wait_for_prompt(b'OK', b'ERROR', store=b'\r\n')
```
Should it be `store=b'\r\n'` or `store=b'%CMNG:'`?

jayteemo replied:
perhaps, i don't remember how that works...

glarsennordic commented on python/modem-firmware-1.3+/device_credentials_installer.py:
```diff
@@ -699,5 +712,63 @@ def main():
+
+def parse_sha(cmng_result_str):
+    try:
+        return cmng_result_str.split('"')[1]
```
This function feels fragile to me

If I understand correctly, you're, on some level, using presence of quotes to detect whether or not the sha was found

I would recommend instead, perhaps:
```py
def parse_at_resp(at_response):
   return at_response.split(',')
```

And then extracting the third argument of that if it exists + some other more direct check of whether the cert requested actually exists in the first place

jayteemo replied:
example response:
`%CMNG: 16842753,2,"4F7BAA4897065359990D982286C1186F0AEA3ADC0C992F8939075150A4D14E81"`
the SHA is in quotes.

glarsennordic replied:
Yeah, I understand -- I'm just saying that that response is comma-separated data, so it might make more sense to someone reading your code if you parse it first by splitting the data by commas

Also I think your current version also detects whether the cert was even present by checking whether or not there was a SHA in the response, (in turn by checking whether `"` is in the response), and that might also throw off a reader

jayteemo committed:
add mfw version checks

script execution should be 1.3.0 or greater and verify option should be 1.3.2 or greater

jayteemo commented:
did some cleanup and added checking of mfw version in an additional commit.
the script should requires at least 1.3.0 and SHA verification requires at least 1.3.2.

plskeggs reviewed (approved):
LGTM.  Nice feature.

junqingzou commented:
Please merge this PR in time, thanks.
